### DISCUSSION TITLE
Eccessiva occupazione del disco

### DISCUSSION THREAD

**Discussion by Dregi56** [indomus/forum#299] [HUB: Home Assistant]
Ho dovuto reinstallare e ripristinare Hassio a causa della mancanza di spazio su disco. Per monitorare il sistema ho visualizzato in Lovelance un sensore riguardante l'uso del disco. Ebbene l'occupazione del disco continua ad aumentare nonostante io non installi nulla di nuovo. Non parlo di poco eh? Solo 48 ore fa registravo una occupazione del disco del 30%, oggi sono già al 60% 
Qualcuno ha qualche idea?

**Comment by Newreader** [comment]
Prova a leggere, se non lo hai già fatto, questo articolo:
https://indomus.it/formazione/configurare-correttamente-recorder-history-e-logbook-su-home-assistant/
che potrebbe darti spunti per la risoluzione del tuo problema

**Reply by Dregi56** [reply]
Mooolto interessante! Grazie. 
Con questa configurazione minimale senza alcun _exclude_ o _include_ potrei avere problemi?:
```
recorder:
  purge_keep_days: 4
```
Come scrivevo, l'uso del disco è arrivato a 70 Gb, ho quidi necessità di _pulire_  un po', ho letto che qualcuno suggerisce anche una cancellazione del file di database perché tanto _verrebbe ricreato all'avvio_ è vero? Mi darebbe problemi?

**Comment by essegi65** [comment]
Inoltre verifica anche i sensori di: temperatura, consumi elettrici, ecc. in quanto registrano diversi dati frequentemente.
In questo caso puoi dare un'occhiata qui: [(https://indomus.it/guide/ridurre-le-scritture-dati-di-sensori-ipertrofici-su-home-assistant/)]

**Reply by Dregi56** [reply]
Però ho controllato: e il file di database home-assistant_v2.db è di 21.228 Kb, non mi sembra granché
C'è invece nella directory config un file strano HNN2DU~T che è di 173.556, nemmeno questo quindi di dimensioni da destare preoccupazione.... Ci sono poi un altro paio di file che hanno l'aria di essere "copie di salvataggio" 
home-assistant_v2.db-shm
home-assistant_v2.db-wal
dei quali non conosco il significato.
Io parlavo di una occupazione crescente e rapida del disco, non  mi sembra che le dimensioni sopra citate siano la causa, correggetemi se sbaglio....

**Reply by Newreader** [reply]
I due file d te menzionati sono due file temporanei  creati da SQLite e legati alla gestione del database. La cancellazione del file home-assistant_v2.db non crea nessun problema perché effettivamente viene ricreato. Ovviamente perdi lo storico.
Hai bisogno, secondo me, di verificare attraverso un software ad hoc (che ti permetta di navigare all'interno della struttura del disco) il contenuto del disco (in termini di directory, file e relative dimensioni) e verificare che cosa determina questo aumento repentino del disco.
Personalmente non utilizzo Hassio ma Home Assistant Core su Raspbian e il protocollo SSH attraverso il software Putty (Windows) per collegarmi in remoto al RPI.  Inoltre uso FileZilla client per vedere la struttura del mio SSD collegato al Raspberry e per trasferire file da e verso il Raspberry.

**Reply by Dregi56** [reply]
Ho ipotizzato che solo un'automazione possa creare un problema di questo tipo. In particolare ho pensato a Home Assistant Google Drive Backup. Penso che questa integrazione, che si occupa di salvare sul cloud periodicamente una immagine di Hassio debba creare un file tar su disco prima di effettuare l'upload. Un file di questo tipo se non viene cancellato occupa uno spazio considerevole su disco. Intanto nella configurazione di questo componente aggiuntivo ho portato a 10 i giorni per la creazione di un nuovo snapshot. Esiste anche una voce da flaggare che riguarda l'eliminazione  dello snapshot salvato sul disco dopo che è stato _uploadato_ ma sembra rispondere poco....sto vedendo.

**Reply by Dregi56** [reply]
Come software per esplorare il disco non saprei cosa usare. Anch'io uso un FTP client per l'eventuale trasferimento di file. Attraverso quest'ultimo ho cercato un'eventuale directory Temp che potrebbe ospitare i file incriminati....ma nulla trovo dappertutto piccole dimensioni.

**Reply by Newreader** [reply]
Se non lo hai fatto ti consiglio di leggere questo README.md su hassio-google-drive-backup, penso che ti chiarirai molte idee e troverai le risposte alle tue domande:
https://github.com/sabeechen/hassio-google-drive-backup
Il README.md, chiaramente, va letto tutto e on molta attenzione.

**Reply by Dregi56** [reply]
Newreader, ti ringrazio per i suggerimenti e per la pazienza. Quello che mi hai suggerito mi è stato utile, anche se alla fine ho scoperto l'arcano:
Un **sensore di movimento** di una telecamera su MotionEye rilevava il movimento delle foglie di un albero col vento e produceva file di discrete dimensioni.

**Reply by Newreader** [reply]
Bene. L'importante è risolvere.
